Vagrant environment does not load ansible.windows modules correctly

### What You Are Seeing?

Specifically in the Vagrant environment, running the tests seems to cause a few errors about `ansible.windows` modules like `win_command` and `win_file` to show up. These don't appear to be problematic in CI, so it may be something with the Vagrant environment specifically or the boxes we're using.

Errors noted include:

```
couldn't resolve module/action 'win_command'. This often indicates a misspelling, missing collection, or incorrect module path

couldn't resolve module/action 'win_file'. This often indicates a misspelling, missing collection, or incorrect module path
```

I tested both with and without fully qualified module syntax (`ansible.windows.<name>`) and both cases failed, which leads me to believe there's something else going on, and it only shows up in Vagrant environment for some reason.

### What is Expected?

Vagrant should run all the tests, the same as the CI runner.

### How Did You Get This To Happen? (Steps to Reproduce)

Run the `build/Invoke-CollectionTests.ps1` script.

Hi: Have you tried to increase the amount of memory for the Vagrant VM? The VM created by Vagrant only has 1 GB of memory. The VM used in Azure CI is a 'Standard_DS2_v2' which has 7 GB of Ram.  I also use Vagrant to run the test of the Chocolatey collection and I experienced random compiler errors on a 4GB Windows 2016 VM. Those errors disappeared when I increased the memory to 8 GB and I managed to run all tests successfully. 

My errors:
`fatal: [win10]: FAILED! => {"changed": false, "msg": "internal error: failed to run exec_wrapper action module_powershell_wrapper: Failed to compile C# code: \r\nerror CS0583: Internal Compiler Error (0xc0000005 at address 00007FF60B64AE7F): likely culprit is 'BIND'.\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(1017,21) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(1002,17) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(1001,22) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(997,22) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(995,17) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(993,13) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(992,13) : error CS0585: Internal Compiler Error: stage 'BIND'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(986,22) : error CS0584: Internal Compiler Error: stage 'BIND' symbol 'Ansible.Basic.AnsibleModule.CheckUnsupportedArguments(System.Collections.IDictionary, System.Collections.Generic.List<string>)'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(986,22) : error CS0584: Internal Compiler Error: stage 'COMPILE' symbol 'Ansible.Basic.AnsibleModule.CheckUnsupportedArguments(System.Collections.IDictionary, System.Collections.Generic.List<string>)'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(33,18) : error CS0584: Internal Compiler Error: stage 'COMPILE' symbol 'Ansible.Basic.AnsibleModule'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(31,11) : error CS0584: Internal Compiler Error: stage 'COMPILE' symbol 'Ansible.Basic'\r\nc:\\Users\\vagrant\\AppData\\Local\\Temp\\x1shvnal.0.cs(31,11) : error CS0584: Internal Compiler Error: stage 'COMPILE' symbol 'Ansible'\r\nerror CS0584: Internal Compiler Error: stage 'COMPILE' symbol '<global namespace>'\r\nerror CS0586: Internal Compiler Error: stage 'COMPILE'\r\nerror CS0587: Internal Compiler Error: stage 'COMPILE'\r\nerror CS0587: Internal Compiler Error: stage 'EMIT'\r\nerror CS0587: Internal Compiler Error: stage 'BEGIN'"}
`

--- a/build/Vagrantfile
+++ b/build/Vagrantfile
@@ -66,6 +66,17 @@ Vagrant.configure("2") do |config|
       # linked clones for speed and size
       vbox.linked_clone = true if Vagrant::VERSION >= '1.8.0'
     end
+
+    client.vm.provision "shell" do |ps|
+      ps.inline = <<-PS1
+        # Configure WinRM for Ansible
+        $installScript = Invoke-WebRequest -Uri "https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1" -UseBasicParsing
+        & ([scriptblock]::Create($installScript))
+        # Enable CredSSP/https in WinRM
+        Enable-WSManCredSSP -Role Server -Force
+      PS1
+      ps.privileged = true
+    end
   end
 
   # Disable automatic box update checking. If you disable this, then

--- a/build/vagrant-dependencies.sh
+++ b/build/vagrant-dependencies.sh
@@ -26,7 +26,7 @@ pip3 install packaging
 pip3 install "Jinja2<3.1.0"
 pip3 install "$ANSIBLE_PACKAGE"
 
-ansible-galaxy collection install ansible.windows
+sudo ansible-galaxy collection install ansible.windows
 
 pip3 --version
 ansible --version

--- a/chocolatey/requirements.txt
+++ b/chocolatey/requirements.txt
@@ -1,2 +1,1 @@
-pyOpenSSL<22.0.0
 pywinrm

--- a/chocolatey/tests/integration/targets/win_chocolatey/tasks/tests.yml
+++ b/chocolatey/tests/integration/targets/win_chocolatey/tasks/tests.yml
@@ -750,5 +750,6 @@
     state: absent
 
 - name: ensure we can still upgrade packages with chocolatey package files missing
-  name: chocolatey
-  state: upgrade
+  win_chocolatey:
+    name: chocolatey
+    state: upgrade

